### ISSUE TITLE
Using a number (i.e. 42) for a category causes error

### ISSUE THREAD

**Issue by coolaj86** [ruhoh/ruhoh.rb#98]
```
ruhoh post "The Meaning of Life"
vim posts/the-meaning-of-life.md
```

Change category to to `42`.

```
ruhoh compile

/Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/gems/ruhoh-1.1/lib/ruhoh/db.rb:28:in `rescue in update': Data type: 'posts' is not a valid data type. (NameError)
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/gems/ruhoh-1.1/lib/ruhoh/db.rb:23:in `update'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/gems/ruhoh-1.1/lib/ruhoh/db.rb:44:in `block in update_all'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/gems/ruhoh-1.1/lib/ruhoh/db.rb:43:in `each'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/gems/ruhoh-1.1/lib/ruhoh/db.rb:43:in `update_all'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/gems/ruhoh-1.1/lib/ruhoh/program.rb:63:in `compile'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/gems/ruhoh-1.1/lib/ruhoh/client/client.rb:139:in `compile'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/gems/ruhoh-1.1/lib/ruhoh/client/client.rb:28:in `initialize'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/gems/ruhoh-1.1/bin/ruhoh:38:in `new'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/gems/ruhoh-1.1/bin/ruhoh:38:in `<top (required)>'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/bin/ruhoh:19:in `load'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/bin/ruhoh:19:in `<main>'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/bin/ruby_noexec_wrapper:14:in `eval'
  from /Users/coolaj86/.rvm/gems/ruby-1.9.3-p194/bin/ruby_noexec_wrapper:14:in `<main>'
```

Change category to `"42"`.

```
ruhoh compile
(works)
```

**Comment by jellelicht** [comment]
I might be misunderstanding something here, but why would you _ever_ want to have a number as the name of a category?

**Comment by coolaj86** [comment]
Well, for example, I want a category called "42" for my philosophical rants about Life, the Universe, and Everything.

Besides, what I want to use it for doesn't validate or invalidate the fact that there's a bug.

**Comment by coolaj86** [comment]
Or maybe I'd like a category 0 to blog about all the null pointer exceptions I encounter. Or 777 for my gambling escapades. Or 1337 for the things that are awesome about me.

Or maybe I'm really into Sesame St and The Count really gets me on a role with today's episode sponsored by the number 12.

The possibilities are almost as limitless as the numbers.

**Comment by plusjade** [comment]
in ruby fixnums cannot be hash keys.   I can automatically use .to_s  on all category names but i then you are still left with the expectation that integers work when in fact the system has mutated them to strings.  so trying to access the category via a number would fail.  since the category hash is just a primitive there isn't a good way for ruhoh to silently handle type conversion. 

long story short the principle of least surprise leans me toward requiring the user pass in strings.

**Comment by coolaj86** [comment]
Then the error message should be changed then. It was very surprising to
find that that long error was because of the category name.

Sent from my iPhone

On Nov 5, 2012, at 10:07 AM, Jade notifications@github.com wrote:

in ruby fixnums cannot be hash keys. I can automatically use .to_s on all
category names but i then you are still left with the expectation that
integers work when in fact the system has mutated them to strings. so
trying to access the category via a number would fail. since the category
hash is just a primitive there isn't a good way for ruhoh to silently
handle type conversion.

long story short the principle of least surprise leans me toward requiring
the user pass in strings.

—
Reply to this email directly or view it on
GitHubhttps://github.com/ruhoh/ruhoh.rb/issues/98#issuecomment-10078713.

**Comment by plusjade** [comment]
yes you are definitely right about that!

**Comment by plusjade** [comment]
The code that caused inaccurate error messages above has been removed in ruhoh 2.x
